packages/plugins/EVM/src/state/TransactionFormatter/descriptors/ITO.ts

```diff
@@ -1,36 +1,79 @@
 import { i18NextInstance } from '@masknet/shared-base'
-import { TransactionContext, isSameAddress, formatBalance } from '@masknet/web3-shared-base'
+import { TransactionContext, isSameAddress } from '@masknet/web3-shared-base'
 import { ChainId, getITOConstants, TransactionParameter } from '@masknet/web3-shared-evm'
+import ITO_ABI from '@masknet/web3-contracts/abis/ITO2.json'
 import { Web3StateSettings } from '../../../settings'
 import type { TransactionDescriptor } from '../types'
+import type { AbiItem } from 'web3-utils'
+import { DescriptorWithTransactionReceipt, getTokenAmountDescription } from '../utils'
 
-export class ITODescriptor implements TransactionDescriptor {
+export class ITODescriptor extends DescriptorWithTransactionReceipt implements TransactionDescriptor {
+    async getClaimTokenInfo(chainId: ChainId, contractAddress: string | undefined, hash: string | undefined) {
+        const connection = await Web3StateSettings.value.Connection?.getConnection?.({
+            chainId,
+        })
+        const events = await this.getReceipt(chainId, contractAddress, ITO_ABI as AbiItem[], hash)
```

Comment:
Better to avoid passing down the hash. We could request the transaction receipt and and it into the context.

Check more: packages/plugins/EVM/src/state/TransactionFormatter.ts

```ts
const receipt = await getTransactionReceipt(/* ... */)
const context = {
    chainId,
    from,
    to,
    value,
    hash,
    receipt,
}
```

Reply:
I don't think get receipt for every transactions is a good idea, since there're some descriptors do not rely on receipt.

Reply:
LGTM